File: src/korean_practice/scenarios/scratchpad.py
"""STT Scratchpad — speak and see transcribed text, no evaluation."""

from dataclasses import dataclass

from korean_practice.scenarios import Scenario, register


@register
@dataclass
class ScratchpadScenario(Scenario):
    id: str = "scratchpad"
    unit: int = 0
    title: str = "STT Scratchpad"
    grammar: list = None
    example_conversations: list = None

    def __post_init__(self):
        if self.grammar is None:
            self.grammar = []
        if self.example_conversations is None:
            self.example_conversations = []

    def briefing(self) -> dict:
        return {
            "id": self.id,
            "unit": self.unit,
            "title": self.title,
            "grammar": [],
            "context": {},
            "scratchpad": True,
            "start_hint": "Hold Space and speak — your words will appear as text.",
        }

File: src/scenarios/scratchpad.ts
/** STT Scratchpad — speak and see transcribed text, no evaluation. */

import { type Scenario, type ScriptStep, type Briefing, type ScenarioContext, register } from "./index";

function createScratchpadScenario(role: string): Scenario {
  return {
    id: "scratchpad",
    unit: 0,
    title: "STT Scratchpad",
    grammar: [],
    role,
    context: {},
    exampleConversations: [],

    roles() { return []; },
    roleDisplayTitle() { return this.title; },
    setup() {},
    conversationScript() { return []; },
    learnerSpeaker() { return "A"; },
    vocabSection() { return ""; },
    formatExamples() { return ""; },

    briefing(): Briefing {
      return {
        id: this.id,
        unit: this.unit,
        title: this.title,
        grammar: [],
        context: {},
        scratchpad: true,
        start_hint: "Hold Space and speak — your words will appear as text.",
      };
    },
  };
}

register(createScratchpadScenario);
